starter-code/src/viking.js

```diff
@@ -1,11 +1,94 @@
 // Soldier
-function Soldier() {}
+function Soldier(healthArg, strengthArg) {
+    this.health = healthArg;
+    this.strength = strengthArg;
+    this.attack = function () {
+        return this.strength;
+    }
+    this.receiveDamage = function (damage) {
+        this.health = this.health - damage;
+    }
+}
 
 // Viking
-function Viking() {}
+function Viking(name, healthArg, strengthArg) {
+    this.name = name;
+    Soldier.call(this, healthArg, strengthArg);
+    this.receiveDamage = function (damage) {
+        this.health = this.health - damage;
+        if (this.health <= 0) {
+            return name + " has died in act of combat";
+        } else {
+            return name + " has received " + damage + " points of damage";
+        }
+    };
+    this.battleCry = function () {
+        return "Odin Owns You All!";
+    };
+}
+
+Viking.prototype = Object.create(Soldier.prototype);
+Viking.prototype.constructor = Viking;
 
 // Saxon
-function Saxon() {}
+function Saxon(healthArg, strengthArg) {
+    Soldier.call(this, healthArg, strengthArg);
+    this.receiveDamage = function (damage) {
+        this.health = this.health - damage;
+        if (this.health <= 0) {
+            return "A Saxon has died in combat";
+        } else {
+            return "A Saxon has received " + damage + " points of damage";
+        }
+    };
+}
+
+Saxon.prototype = Object.create(Soldier.prototype);
+Saxon.prototype.constructor = Saxon;
 
 // War
-function War() {}
+function War() {
+    this.vikingArmy = [];
+    this.saxonArmy = [];
+    this.addViking = function (viking) {
+        this.vikingArmy.push(viking);
+    };
+    this.addSaxon = function (saxon) {
+        this.saxonArmy.push(saxon);
+    };
+    this.vikingAttack = function () {
+        var vikingFighter = this.vikingArmy[Math.floor(Math.random()*this.vikingArmy.length)];
+        var saxonDamaged = this.saxonArmy[Math.floor(Math.random()*this.saxonArmy.length)];
+        var fightResult = saxonDamaged.receiveDamage(vikingFighter.strength);
```

Comment:
Don't access the viking strength directly, you have a method for that: attack().